core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -163,6 +177,18 @@ public abstract void handleSubchannelState(
 
   /**
    * A balancing decision made by {@link SubchannelPicker} for an RPC.
+   *
+   * <p>The outcome of the decision will be one of the following:
+   * <ul>
+   *   <li>Proceed: if a Subchannel is provided via {@link #withSubchannel}, and is in READY state
+   *   when the RPC tries to start on it, the RPC will proceed on that Subchannel.</li>
+   *   <li>Error: if an error is provided via {@link #withError}, and the RPC is not wait-for-ready
+   *   (i.e., {@link CallOptions#withWaitForReady} was not called), the RPC will fail immediately
+   *   with the given error.</li>
+   *   <li>Buffer: in all other cases, the RPC will be buffered in the Channel, until next picker is
```

Comment:
Question: What is a good way for a caller to distinguish "Proceed" and "Buffer". My guess was that the caller can check the state of `subchannel`. Is that correct?